commcare_connect/opportunity/views.py

```diff
@@ -633,10 +635,20 @@ def payment_import(request, org_slug=None, opp_id=None):
     if file_format not in ("csv", "xlsx"):
         raise ImportException(f"Invalid file format. Only 'CSV' and 'XLSX' are supported. Got {file_format}")
 
+    redirect_url = reverse("opportunity:worker_payments", args=(org_slug, opp_id))
+
+    lock = cache.lock(get_payment_upload_key(request.opportunity.pk))
+    got_lock = lock.acquire(blocking=False)
+
+    if not got_lock:
+        messages.error(request, "Another payment import is in progress. Please try again later.")
+        return redirect(f"{redirect_url}?{request.GET.copy().urlencode()}")
+
+    lock.release()
```

Comment:
Why do you release the lock here? Don't we want this import to have the lock at this stage until the import is done?

Reply:
Oh, I see we lock it again in the task